curriculum/challenges/english/25-front-end-development/lab-bank-account-manager/6718d2d59337c822ecb697ff.md

```diff
@@ -0,0 +1,275 @@
+---
+id: 6718d2d59337c822ecb697ff
+title: Build a Bank Account Management Program
+challengeType: 14
+dashedName: build-a-bank-account-management-program
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should define a class named `BankAccount` with a constructor that takes `balance` as parameter. The constructor should set the default `balance` to `0`and initialize an empty array named `transactions` to store transaction records as objects. 
+
+2. The `transactions` object defined inside the `BankAccount` constructor should be able to take two properties: `type` and `amount`. The `type` property should be either `deposit` or `withdraw`, and the `amount` property should be the amount deposited or withdrawn.
+
+3. Inside the `BankAccount` class, You should define a method named `updateBalance` that updates the balance according to the type of transaction. If the transaction type is `deposit`, it should add the amount to the balance. If the transaction type is `withdraw`, it should subtract the amount from the balance.
+
+4. You should define a method named `deposit` that takes the deposit amount as a parameter. This method should update the current balance according to deposits. When the deposit amount is greater than `0`, it should:
+
+   - Push a new object to the transactions array with a type of deposit and the amount deposited.
+   - Update the balance.   
+   - Return `"Successfully deposited $[amount]. New balance: $[balance]"`.
+
+5. In the `deposit` method, if the amount is less than or equal to `0`, it should return `"Deposit amount must be greater than zero."`.
+
+6. You should define a method named `withdraw` that takes an amount as a parameter. This method should update the current balance according to withdrawals. When the amount to be withdrawn is greater than `0` and less than or equal to the current balance, it should:
+
+   - Push a new object to the transactions array with a type of withdraw and the amount withdrawn.
+   - Update the balance.
+   - Return `"Successfully withdrew $[amount]. New balance: $[balance]"`. 
+
+7. In the `withdraw` method, when the amount to be withdrawn is less than or equal to `0` or greater than the current balance, it should return `"Insufficient balance or invalid amount."`.
+
+
+8. You should define a method named `checkBalance` that returns the current balance in the format `"Current balance: $[balance]"`.
```

Comment:
```suggestion
8. You should define a method named `checkBalance` that returns the current balance in the format `"Current balance: ${balance}"`.
```

Reply:
I have used square brackets as placeholders here and not as template literals.

Reply:
```suggestion
8. You should define a method named `checkBalance` that returns the current balance in the format `"Current balance: $<balance>"`.
```

I have seen in the curriculum most often the angular brackets to signal placeholder

Reply:
I am a bit confused but let's go with the optimal option @zairahira